File: frontend/src/types/magnet.ts
export interface MagnetFile {
  path: string;
  size: number;
  index: number;
}

export interface MagnetLink {
  id: number;
  hash: string;
  title: string;
  size: number;
  files: MagnetFile[];
  created_by: number;
  created_at: string;
}

export interface MagnetLinkList {
  links: MagnetLink[];
  total: number;
}

export interface StreamInfo {
  stream_url: string;
  file_name: string;
  file_size: number;
  mime_type: string;
  duration?: number;
}

export interface MagnetLinkParseResponse {
  magnet_link: MagnetLink;
  stream_info?: StreamInfo;
}

File: backend/magnet_api/app/schemas/magnet.py
from typing import List, Optional
from pydantic import BaseModel, Field
from datetime import datetime

class MagnetLinkCreate(BaseModel):
    magnet_url: str = Field(..., description="Magnet URL to parse")

class MagnetFile(BaseModel):
    path: str
    size: int
    index: int

class MagnetLink(BaseModel):
    id: int
    hash: str
    title: str
    size: int
    files: List[MagnetFile]
    created_by: int
    created_at: datetime = Field(default_factory=datetime.utcnow)
    
    class Config:
        from_attributes = True

class MagnetLinkList(BaseModel):
    links: List[MagnetLink]
    total: int

class StreamInfo(BaseModel):
    stream_url: str
    file_name: str
    file_size: int
    mime_type: str
    duration: Optional[float] = None
